extensions/amp-story/1.0/amp-story-embedded-component.js

```diff
@@ -849,21 +858,18 @@ export class AmpStoryEmbeddedComponent {
         const pageRect = pageEl./*OK*/ getBoundingClientRect();
         const elRect = element./*OK*/ getBoundingClientRect();
 
-        if (elRect.width >= elRect.height) {
-          state.newWidth = pageRect.width;
-          state.scaleFactor = elRect.width / state.newWidth;
-          state.newHeight = (elRect.height / elRect.width) * state.newWidth;
-        } else {
-          const maxHeight = pageRect.height - VERTICAL_PADDING;
-          state.newWidth = Math.min(
-            (elRect.width / elRect.height) * maxHeight,
-            pageRect.width
-          );
-          state.newHeight = (elRect.height / elRect.width) * state.newWidth;
-          state.scaleFactor = elRect.height / state.newHeight;
-        }
+        // If screen is very wide and story has supports-landscape attribute,
+        // we don't want it to take the whole width. We take the maximum width
+        // that the embed can actually take instead.
+        state.newWidth = Math.min(pageRect.width, MAX_EMBED_WIDTH_PX);
+
+        // Since we don't know the actual width of the content inside the iframe
+        // and in responsive environments the iframe takes the whole width. We
+        // hardcode a limit based on how we know the content behaves. See
+        // #22334.
```

Comment:
This comment might be more useful with the `MAX_EMBED_WIDTH_PX` declaration. You might want to add that this is only true for Twitter :))